app/harvesters/hathi_harvester.rb

```diff
@@ -0,0 +1,94 @@
+require 'net/http'
+require 'rubygems/package'
+require 'zlib'
+
+
+##
+# A harvester for Hathi Trust MARC XML dump files.
+#
+# Supports either harvesting a previously downloaded dump file, or downloading
+# the latest available dump file and harvesting that.
+class HathiHarvester < Krikri::Harvesters::MarcXMLHarvester
+
+  ##
+  # @see Krikri::Harvesters::MarcXMLHarvester#each_collection
+  def each_collection
+    tarball_source do |tar_file|
+      each_tar_entry(tar_file) do |entry|
+        yield entry
+      end
+    end
+  end
+
+  private
+
+  ##
+  # @yield [IO] Give an IO stream containing an archive of MARC XML collection
+  #   files in TAR format.
+  def tarball_source(&block)
+    if File.exists?(uri)
+      File.open(uri, &block)
+    else
+      download_url(get_latest_tarball_url, &block)
+    end
+  end
+
+  ##
+  # @return [URI] Determine the URL of the most recent Hathi data dump.
+  def get_latest_tarball_url
+    html = Net::HTTP.get(URI(uri))
+    parsed_html = Nokogiri::HTML(html)
+
+    file_names = parsed_html.xpath('//a[starts-with(@href, "dpla")]').map { |a| a.get_attribute("href") }
+    latest_file = file_names.sort.reverse.first
+
+    URI.join(uri, latest_file)
+  end
+
+  ##
+  # Download the contents of a URL to a temporary file.
+  #
+  # @yield [File] If a block is given, the temporary file will be provided and
+  #   removed when the block returns.
+  # @return [File] If no block is given, return the temporary file.  The caller
+  #   must call #unlink once finished with the file.
+  def download_url(url)
+    Net::HTTP.start(url.host, url.port) do |http|
+      request = Net::HTTP::Get.new(url.request_uri)
+
+      file = Tempfile.new('hathi', Dir.tmpdir, :encoding => 'ASCII-8BIT')
```

Comment:
I guess I'm unclear as to why an encoding has to be specified if it's a binary file that's going to be written to the tempfile.  It's going to be a `.tar.gz` file.  Is ASCII-8BIT specified to make sure it gets treated as a stream of bytes?